sensors: introduce kconfig named choices

erwango opened the pull request:
Use Kconfig specific object "named choices".
Aim is to allow to define config choices selection in Kconfig.*
files instead of _defconfig and hence allow to keep flags
activation conditional.

When not specified, default choice value is the first option defined.
Hence, options order have been changed to keep default choices
default value unchanged.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@galak , can we merge this one?
It solves 2 annoygin warning messages when building some boards:
```
warning: HTS221_TRIGGER_NONE (defined at drivers/sensor/hts221/Kconfig:42) was assigned the value
'y' but got the value 'n'. You can check symbol information (including dependencies) in the
'menuconfig' interface (see the Application Development Primer section of the manual), or in the
Kconfig reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_HTS221_TRIGGER_NONE.html
(which is updated regularly from the master branch). See the 'Setting configuration values' section
of the Board Porting Guide as well.
warning: LIS3MDL_TRIGGER_NONE (defined at drivers/sensor/lis3mdl/Kconfig:52) was assigned the value
'y' but got the value 'n'. You can check symbol information (including dependencies) in the
'menuconfig' interface (see the Application Development Primer section of the manual), or in the
Kconfig reference at
http://docs.zephyrproject.org/reference/kconfig/CONFIG_LIS3MDL_TRIGGER_NONE.html (which is updated
regularly from the master branch). See the 'Setting configuration values' section of the Board
Porting Guide as well.
```

erwango commented:
^^ @galak

ulfalizer reviewed (dismissed):
Are those `.config` file assignments meant to set the value of the choice btw, or just give it a default?

Previously, you would get a warning if you "overrode" a choice selection from a board defconfig file in `prj.conf` (by setting some other choice symbol to `y`, which indirectly gives all the other choice symbols the value `n`). That's been fixed in `scripts/kconfig/kconfig.py` now though, specifically to allow overriding.

erwango committed:
sensors: introduce kconfig named choices

Use Kconfig specific object "named choices".
Aim is to allow to define config choices selection in Kconfig.*
files instead of _defconfig and hence allow to keep flags
activation conditional.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented on drivers/sensor/hts221/Kconfig:
```diff
@@ -29,16 +29,12 @@ config HTS221_I2C_MASTER_DEV_NAME
+choice HTS221_TRIGGER_MODE
 	prompt "Trigger mode"
 	depends on HTS221
-	default HTS221_TRIGGER_GLOBAL_THREAD
```
Indeed. Updated wrt your advice. Thanks

erwango commented on boards/arm/disco_l475_iot1/Kconfig.defconfig:
```diff
@@ -94,6 +94,23 @@ config PWM_STM32_2
+if LIS3MDL
+
+choice LIS3MDL_TRIGGER_MODE
+	default LIS3MDL_TRIGGER_NONE
```
Done thanks

erwango commented:
@ulfalizer : Thanks for the review. Change updated
